.github/workflows/plan-terraform.yaml

```diff
@@ -18,18 +18,22 @@ jobs:
       matrix:
         test:
         - name: Openstack
-          yaml: test/openstack.yaml
+          yaml: ./openstack.yaml
           cloud_integration: openstack
         - name: Vanilla
-          yaml: test/vanilla.yaml
+          yaml: ./vanilla.yaml
           cloud_integration: ''
     env:
-      TF_VAR_model: test
+      TF_VAR_branch: ${{ github.head_ref }}
       TF_VAR_manifest_yaml: ${{ matrix.test.yaml }}
       TF_VAR_cloud_integration: ${{ matrix.test.cloud_integration }}
-      WORKING_DIR: 'terraform'
+      WORKING_DIR: 'terraform/test'
     steps:
     - uses: actions/checkout@v4
+      with:
+        fetch-depth: 0
+        sparse-checkout: ${{env.WORKING_DIR}}
+        sparse-checkout-cone-mode: false
```

Comment:
@asbalderson here is where the GH action checks out only the `terraform/test` directory to test in "isolation"